trace-mini-agent/src/mini_agent.rs

```diff
@@ -69,29 +98,31 @@ impl MiniAgent {
     async fn trace_endpoint_handler(
         req: Request<Body>,
         trace_processor: Arc<dyn trace_processor::TraceProcessor + Send + Sync>,
-        tx: Sender<pb::TracerPayload>,
+        trace_tx: Sender<pb::TracerPayload>,
+        stats_processor: Arc<dyn stats_processor::StatsProcessor + Send + Sync>,
+        stats_tx: Sender<pb::ClientStatsPayload>,
     ) -> http::Result<Response<Body>> {
         match (req.method(), req.uri().path()) {
             (&Method::PUT, TRACE_ENDPOINT_PATH) => {
-                match trace_processor.process_traces(req, tx).await {
+                match trace_processor.process_traces(req, trace_tx).await {
+                    Ok(res) => Ok(res),
+                    Err(err) => log_and_return_http_error_response(&format!(
```

Comment:
maybe we can pass the str + err and format it in log_and_return_http_error_response?